harness/determined/experimental/pytorch_lightning/adapter.py

```diff
@@ -0,0 +1,85 @@
+from typing import Callable, NewType, Any
+import pytorch_lightning as ptl
+from determined.pytorch import PyTorchTrial, PyTorchTrialContext
+from determined.experimental.pytorch_lightning.data_module import DETLightningDataModule
+from determined import monkey_patch
+from typing import Any, Dict, Sequence, Union
+from inspect import signature
+import torch
+import sys
+
+TorchData = Union[Dict[str, torch.Tensor], Sequence[torch.Tensor], torch.Tensor]
+HyperparamsProvider = Callable[[str], Any]
+
+
+def bail(msg: str = '', fail: bool = True):
+    msg = f'NotSupported: {msg}'
+    if fail:
+        raise TypeError(msg)
+    else:
+        # TODO use a logger
+        print(msg, file=sys.stderr)
+
+
+class DETLightningModule(ptl.LightningModule):
```

Comment:
Why do we need a separate class for it? We should take an instantiated lightning module from users rather than defining a separate lightning module as we discussed last time